playbooks/allocate_pubcloud.yml

```diff
@@ -2,63 +2,21 @@
 - hosts: localhost
   connection: local
   gather_facts: False
+  vars:
+    boot_success: false
   tasks:
-    - name: Provision a set of public cloud instances
-      local_action:
-          module: rax
-          name: "{{ name }}"
-          flavor: "{{ flavor }}"
-          image: "{{ image }}"
-          count: "{{ count }}"
-          key_name: "{{ keyname }}"
-          region: "{{ region }}"
-          wait: yes
-          wait_timeout: 900
-          auto_increment: no
-      register: rax
-      until: rax|success
-      retries: 120
-      delay: 60
 
-    - name: Display rax output on failure
-      debug:
-        var: rax
-      when:
-        - rax.success|length < count
+    # Retry instance creation, trying each region in turn.
+    # region should be a CSV string.
 
-    - name: Fail on duplicate instance
-      fail:
-        msg: |
-          While nova allows multiple instances with the same name,
-          the ansible rax module will not create an instance if one
-          with the requested name already exists. Please create
-          instances with unique names, using the common.gen_instance_name
-          function
-      when:
-        - rax.success|length < count
-        - rax.instances|length == count
+    # region_retries is the shuffled list of regions repeated 5 times
+    # We itterate over that list attempting to boot the instance(s).
+    # When booting succeeds, boot_success is set to true so that
+    # all subsequent attempts are skipped.
 
-    - name: Fail if the required number of instances aren't available
-      fail:
-        msg: "At least one public cloud instance failed to start :("
-      when: rax.success|length < count
+    - name: Create region retries list
+      set_fact:
+        region_retries: "{{ region.split(',')|map('trim')|shuffle * 5 }}"
 
-    - name: Write inventory
-      copy:
-        content: |
-          [job_nodes:children]
-          hosts
-
-          [hosts]
-          {% for instance in rax.success %}
-          {{instance.name}} ansible_host={{instance.accessIPv4}} ansible_user=root
-          {% endfor %}
-        dest: '{{lookup("env", "WORKSPACE")}}/rpc-gating/playbooks/inventory/hosts'
-
-    - name: Wait for SSH to be available on all hosts
-      wait_for: port=22 host="{{ item.accessIPv4 }}"
-      with_items: "{{ rax.success }}"
-
-    - name: Wait an additional amount of time for any remaning startup tasks to finish
-      pause:
-        minutes: 5
+    - include: tasks/boot_instance_attempt.yml
+      with_items: "{{ region_retries }}"
```

Comment:
Could you add the `when` here, to avoid having to suffix each task in `boot_instance_attempt.yml` with it?